assets/src/js/main.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Main scripts, loaded on all pages.
+ *
+ * @package wp-mentions-links
+ */
+
+/**
+ * Post and User Autocompleter hooks.
+ *
+ * @see https://developer.wordpress.org/block-editor/components/autocomplete/
+ */
+
+const excludedBlocks = [];
+
+const postCompleter = {
+	name: 'posts-completer',
+	triggerPrefix: '#',
+	isDebounced: true,
+	options: async ( search ) => {
+		let queryString = '';
+		if ( search ) {
+			queryString = '?search=' + encodeURIComponent( search );
+		}
+
+		let returnArray = [];
+		if ( 'undefined' !== typeof wpMentionsLinks.supportedCPTs ) {
+			for ( let i in wpMentionsLinks.supportedCPTs ) {
+				returnArray = returnArray.concat( await wp.apiFetch( { path: '/wp/v2/' + wpMentionsLinks.supportedCPTs[i] + '' + queryString } ) );
+			}
+		} else {
+			returnArray = returnArray.concat( await wp.apiFetch( { path: '/wp/v2/posts' + queryString } ) );
+			returnArray = returnArray.concat( await wp.apiFetch( { path: '/wp/v2/pages' + queryString } ) );
+		}
+
+		return returnArray;
+	},
+	getOptionKeywords: ( post ) => post.title.rendered.split( /\s+/ ),
+	getOptionLabel: ( post ) => [
+		<span class="dashicons dashicons-text-page"></span>,
+		<span>&nbsp;{ ( 30 < post.title.rendered.length ? post.title.rendered.substring( 0, 30 ) + '...' : post.title.rendered ) }</span> // Add ellipsis.
```

Comment:
Is this JSX?

Reply:
Please see https://reactjs.org/docs/jsx-in-depth.html

Reply:
Yes this is JSX. I changed `class` to `className` and added `key` on both spans.